### PR TITLE
nixos/ombi: add baseUrl option

### PR DESCRIPTION
**Pull request by Elfahor**
The base URL GUI option of Ombi is known to be broken, which is very inconvenient for reverse proxying with subfolders. The suggested method (in Ombi's documentation) is to pass the `--baseurl` command-line parameter.

## Description of changes

This commit thus adds an option to set this parameter to the NixOS Ombi module.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/misc/ombi.nix [hunk 1]
```diff
@@ -31,6 +31,12 @@ in {
         description = lib.mdDoc "Open ports in the firewall for the Ombi web interface.";
       };
 
+      baseUrl = mkOption {
+        type = types.str;
+        default = "/";
+        description = lib.mdDoc "The base URL for the Ombi web interface.";
```
Comment (nu-nu-ko) [inline nixos/modules/services/misc/ombi.nix:37]: `lib.mdDoc` is now just an alias and can be safely removed everywhere.
see https://github.com/NixOS/nixpkgs/commit/d36f950d405369abca5144b14a201bc25c4a080a and https://github.com/NixOS/nixpkgs/pull/237557

File: nixos/modules/services/misc/ombi.nix [hunk 2]
```diff
@@ -59,7 +65,7 @@ in {
         Type = "simple";
         User = cfg.user;
         Group = cfg.group;
-        ExecStart = "${pkgs.ombi}/bin/Ombi --storage '${cfg.dataDir}' --host 'http://*:${toString cfg.port}'";
+        ExecStart = "${pkgs.ombi}/bin/Ombi --storage '${cfg.dataDir}' --host 'http://*:${toString cfg.port}' --baseurl ${cfg.baseUrl}";
```
Comment (nu-nu-ko) [inline nixos/modules/services/misc/ombi.nix:68]: `${lib.getExe pkgs.ombi}` would be nice.

File: nixos/modules/services/misc/ombi.nix (hunk 1 above)
Comment (Elfahor) [reply nixos/modules/services/misc/ombi.nix:37]: Do you want me to remove it from all the options in the module, or only for the one I added?

File: nixos/modules/services/misc/ombi.nix (hunk 2 above)
Comment (Elfahor) [reply nixos/modules/services/misc/ombi.nix]: Done in [883d86f](https://github.com/NixOS/nixpkgs/pull/284311/commits/883d86fefaa8833e220bb0a9ebc4a41e05c79c26)

File: nixos/modules/services/misc/ombi.nix (hunk 1 above)
Comment (Elfahor) [reply nixos/modules/services/misc/ombi.nix]: I removed everywhere in [106710b](https://github.com/NixOS/nixpkgs/pull/284311/commits/106710b5e1d01d871a527ef17bdd17cdc485df96)

**Comment by Elfahor** [comment]
I have successfully tested these changes on a server.

**Review by fsnkty — APPROVED** [review]
changes seem fine, id be more descriptive in the enable option about what the service is though..

also seen in the module unrelated to these changes, would be nice to see `systemd.tmpfiles.settings` used insteead.

**Comment by Elfahor** [comment]
Should I make these changes in this PR or open an other one?

**Comment by nu-nu-ko** [comment]
> Should I make these changes in this PR or open an other one?

Id assume same PR different commit but am unsure, either way neither are obligations of yours, just random things id like to see personally.

I'd wait for someone more familiar with contributing to get back to you on that one if you wish to make the changes.

**Comment by Elfahor** [comment]
I'm open to making them. I just wish that what I already added be merged, since I added the new `baseUrl` option because I needed it in the first place.

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution! Please rebase to drop the merge commit, as well as to drop the mdDoc removal commit (this was handled treewide). (Of course, the new option shouldn't use lib.mdDoc either, as it was removed from nixpkgs.)

Otherwise the changes look good and I'll be happy to merge afterwards.

**Comment by Elfahor** [comment]
Unfortunately, I don't think this is going to work. Closing.

**Pull request closed without merging**